### PR TITLE
W3 - hieucap - create save data

### PR DESCRIPTION
**Pull request by at-hieucap**
#### Related issues (Optional)
Task #37 

#### Description (Required)
- [x] Share Preference
- [x] External storage
- [x] Database

### CODE REVIEW

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/CompanyActivity.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/CompanyActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/ExternalStorageActivity.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/ExternalStorageActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/PeopleAdapter.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/PeopleAdapter.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/PeopleSQLite.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/PeopleSQLite.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/SaveDataActivity.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/SaveDataActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/SaveDatabaseActivity.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/SaveDatabaseActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/saveData/SharePreferenceActivity.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package vn.asiantech.internship.saveData;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/saveData/SharePreferenceActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

**Comment by at-hieucap** [comment]
@at-kietvo please review for me,
thank you so much!

File: app/src/at-hieucap/java/vn/asiantech/internship/model/Company.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package vn.asiantech.internship.model;
+
+public class Company {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/model/Company.java]: Pls add Javadoc

File: app/src/at-hieucap/java/vn/asiantech/internship/model/People.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
+package vn.asiantech.internship.model;
+
+public class People {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/model/People.java]: Pls add Javadoc

File: app/src/at-hieucap/java/vn/asiantech/internship/model/People.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    private int iDPeople;
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/model/People.java]: pls change to idPeople

File: app/src/at-hieucap/java/vn/asiantech/internship/model/People.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    private String namePeople;
+    private int agePeople;
+
+    public People() {
+    }
+
+    public int getIDPeople() {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/model/People.java]: Change all `ID` -> `Id`

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/CompanyActivity.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.model.Company;
+
+public class CompanyActivity extends AppCompatActivity {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/CompanyActivity.java]: Pls add Javadoc

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/ExternalStorageActivity.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Scanner;
+
+import vn.asiantech.internship.R;
+
+public class ExternalStorageActivity extends AppCompatActivity {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/ExternalStorageActivity.java]: Pls add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.model.People;
+
+public class PeopleAdapter extends RecyclerView.Adapter<PeopleAdapter.PeopleViewHolder> {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+            holder.onBindData(mPeoples.get(position));
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mPeoples.size();
+    }
+
+    public interface OnItemClickListener {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        void onClickItem(int position);
+    }
+
+    static class PeopleViewHolder extends RecyclerView.ViewHolder {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public PeopleViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_people, parent, false);
+        final PeopleViewHolder peopleViewHolder = new PeopleViewHolder(view);
+        peopleViewHolder.mTvNamePeople.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mOnItemClickListener.onClickItem((peopleViewHolder.getAdapterPosition()));
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java]: Pls remove 1 `( )`

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+
+        PeopleViewHolder(View itemView) {
+            super(itemView);
+            mTvIdPeople = itemView.findViewById(R.id.tvIDPeople);
+            mTvNamePeople = itemView.findViewById(R.id.tvNamePeople);
+            mTvSlogan = itemView.findViewById(R.id.tvSlogan);
+        }
+
+        private void onBindData(People people) {
+            mTvIdPeople.setText(String.valueOf(people.getIDPeople()));
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleAdapter.java]: Change `ID` -> `Id`

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleSQLite.java [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import java.util.ArrayList;
+
+import vn.asiantech.internship.model.Company;
+import vn.asiantech.internship.model.People;
+
+public class PeopleSQLite extends SQLiteOpenHelper {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleSQLite.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleSQLite.java [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+        sqLiteDatabase.execSQL(CREATE_TABLE_EMPLOYEE);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
+        sqLiteDatabase.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME_EMPLOYEE);
+        onCreate(sqLiteDatabase);
+    }
+
+    void insertUSER(int id, String name, int age) {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/PeopleSQLite.java]: Pls change USER -> ` User`

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/SaveDataActivity.java [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import vn.asiantech.internship.R;
+
+public class SaveDataActivity extends AppCompatActivity implements View.OnClickListener {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/SaveDataActivity.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/SaveDatabaseActivity.java [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.model.People;
+
+public class SaveDatabaseActivity extends AppCompatActivity implements PeopleAdapter.OnItemClickListener {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/SaveDatabaseActivity.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/savedata/SharePreferenceActivity.java [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import vn.asiantech.internship.R;
+
+public class SharePreferenceActivity extends AppCompatActivity {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/savedata/SharePreferenceActivity.java]: Pls Add Javadoc.

File: app/src/at-hieucap/java/vn/asiantech/internship/ui/main/MainActivity.java [hunk 24]
```diff
@@ -8,29 +8,44 @@
 import android.widget.Button;
 
 import vn.asiantech.internship.R;
+import vn.asiantech.internship.savedata.SaveDataActivity;
 import vn.asiantech.internship.ui.calculator.CalculatorActivity;
-import vn.asiantech.internship.ui.recyclerview.PersonViewActivity;
 import vn.asiantech.internship.ui.login.LoginActivity;
+import vn.asiantech.internship.ui.recyclerview.PersonViewActivity;
 
 public class MainActivity extends AppCompatActivity implements View.OnClickListener {
```
Comment (at-kietvo) [inline app/src/at-hieucap/java/vn/asiantech/internship/ui/main/MainActivity.java]: Pls Add Javadoc.

**Comment by at-hieucap** [comment]
@at-kietvo I fix finished comments , please review for me.
Thank you so much !

**Comment by at-hieucap** [comment]
@at-kietvo I fix finished conflicts , please review for me.
Thank you so much !

**Pull request merged**